### PR TITLE
Add a Code of Conduct

### PR DESCRIPTION
**Pull request by RichardLitt**
This PR adds the Contributor Covenant Code of Conduct. It is the industry standard for open source projects at the moment, used by over 40,000 projects. While there are some issues with code of conducts in general - they can be perceived as threatening, they can suggest active moderation, and they may exclude certain viewpoints (if you're not aware of what I mean, let me know, and I can bring up examples) - they are overwhelmingly supported by communities that view healthy interactions as a core value, and by communities that are aim to be welcoming.

One of the issues with CoCs is that they imply rather than explicitly describe moderation procedures. I want to propose a three strike warning system, after which violators would be excluded from the community space. The moderators in charge would be the OrbitDB contributors - anyone who has push access to OrbitDB repositories. I've added a line to that affect in the Contributing guide, in this PR. If you have suggestions for a better system, let me know.

Another issue with CoCs is the scope. Here, I want to be clear that I am suggesting we use this Code for all of OrbitDB's interactions - on Gitter, on IRC, and in GitHub issues and PRs, and in any other place where OrbitDB moderators would exist (not including something like Twitter, where we don't moderate). If someone who is part of the community starts harassing someone offline or outside of OrbitDB's space, and it impacts the aggrieved's ability to work on OrbitDB, then we enter a gray area. Hopefully, this isn't an issue anytime soon. In such cases (which are not hypothetical - I've seen them happen and destroy members' involvement in projects), I think that we should proactively enforce the code by removing the aggressor from OrbitDB community spaces and especially from any position of power within the OrbitDB community (aka, no push rights.) This process should follow the three-strike process outlined above. 

This all sounds like a hassle. The point is that newcomers know that this is in place, so they feel like the can work on OrbitDB without having to deal with jerks. I think the initial hassle setting it up is worth that. 

I'd like a review from all of the @orbitdb/maintainers before merging. :)

### CODE REVIEW

**Comment by sirfumblestone** [comment]
Looks good to me 👍

File: contributing.md [hunk 1]
```diff
@@ -46,7 +47,11 @@ We want to keep the OrbitDB community awesome, growing and collaborative. We nee
...
+There is also a more extensive [Code of conduct](CODE_OF_CONDUCT.md) which we follow, which can be found as part of the related Haja Networks organization.
+
+#### Moderation
+
+In cases where community members transgress against the values above or in the Code, members of the OrbitDB organization members will use a three-strike warning system, where the aggressor will be warned twice before they are permanently excluded from OrbitDB community spaces. This code applies to Gitter, IRC, and GitHub, and any other future space that OrbitDB uses for communication. For interactions between OrbitDB community members outside of this space, the code also applies if the interactions are reported and deemed to be interfering with community members safely working on OrbitDB together. Moderation conversations, where more serious than simple warnings, will occur in private repositories or by email to ensure anonymity for reporters, and to ensure the safety of the moderators. To report an instance, please see the emails in the [Code of Conduct](CODE_OF_CONDUCT.md).
```
Comment (haadcode) [inline contributing.md]: Small details, but perhaps add "community" to "...and any other future space that OrbitDB *community* uses for communication."

File: contributing.md [hunk 2]
```diff
@@ -46,7 +47,11 @@ We want to keep the OrbitDB community awesome, growing and collaborative. We nee
 
 - Stay on topic: Make sure that you are posting to the correct channel and avoid off-topic discussions. Remember when you update an issue or respond to an email you are potentially sending to a large number of people. Please consider this before you update. Also remember that nobody likes spam.
 
-There is also a more extensive [Code of conduct](/code-of-conduct.md) which we follow, which can be found as part of the related Haja Networks organiaation.
+There is also a more extensive [Code of conduct](CODE_OF_CONDUCT.md) which we follow, which can be found as part of the related Haja Networks organization.
```
Comment (haadcode) [inline contributing.md]: Should this be "OrbitDB community" instead of "Haja Networks"?

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at [community@orbitdb.org](mailto:community@orbitdb.org), which goes to all members of the @OrbitDB community team, or to [richardlitt@orbitdb.org](mailto:richardlitt@orbitdb.org), which goes only to [@RichardLitt](https://github.com/RichardLitt) or to [haadcode@orbitdb.org](mailto:haadcode@orbitdb.org), which goes only to [@haadcode](https://github.com/haadcode).
+
+All
```
Comment (haadcode) [inline CODE_OF_CONDUCT.md]: The line cuts here

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+Examples of behavior that contributes to creating a positive environment
+include:
+
+* Using welcoming and inclusive language
+* Being respectful of differing viewpoints and experiences
+* Gracefully accepting constructive criticism
+* Focusing on what is best for the community
+* Showing empathy towards other community members
+
```
Comment (haadcode) [inline CODE_OF_CONDUCT.md:22]: I really love the examples here! ❤️

**Comment by haadcode** [comment]
This is *awesome* @RichardLitt! Thank you for putting this up. I'm happy we finally have a CoC :)

Small comments, but looks good to me and I'm 100% behind this code of conduct 👍

**Comment by haadcode** [comment]
@vvp @shamb0t please review this and let us know if this is ok.

**Comment by shamb0t** [comment]
This is great @RichardLitt, I'm on board! Only question is what is the process to determine that someone has violated code? Jury? Dictator judge?

File: contributing.md (hunk 1 above)
Comment (RichardLitt) [reply contributing.md]: Done. Good point.

File: CODE_OF_CONDUCT.md (hunk 4 above)
Comment (RichardLitt) [reply CODE_OF_CONDUCT.md:22]: They're not mine - just part of the CoC template. They are good, though. :)

**Comment by RichardLitt** [comment]
Once an instance of a violation is reported, the Community Moderation team would take a look at it and decide if it was a violation. The team would be made up of community contributors in the @orbitdb org who want to be part of it. I'll make a PR to add this, if that answers your question?

**Review by vvp — APPROVED** [review]
I'm definitely ok with this. I appreciate that we specify the scope, I believe it will ease the most serious doubts people have (compared to vanilla Contributor Covenant) :+1:

**Comment by RichardLitt** [comment]
@shamb0t Let me know if I've answered your question! :)

**Comment by RichardLitt** [comment]
Sweet. Thanks all, we now have a Code of Conference. I am going to start referencing this from the other repositories in the organization. :)

**Comment by RichardLitt** [comment]
Sweet. Thanks all, we now have a Code of Conference. I am going to start referencing this from the other repositories in the organization. :)

**Comment by RichardLitt** [comment]
Sweet. Thanks all, we now have a Code of Conference. I am going to start referencing this from the other repositories in the organization. :)

**Comment by RichardLitt** [comment]
Sweet. Thanks all, we now have a Code of Conference. I am going to start referencing this from the other repositories in the organization. :)

**Comment by RichardLitt** [comment]
Sweet. Thanks all, we now have a Code of Conference. I am going to start referencing this from the other repositories in the organization. :)

**Pull request merged**